airflow/dags/sync_ntd_data_xlsx/scrape_ntd_ridership_xlsx_url.py

```diff
@@ -0,0 +1,41 @@
+# ---
+# python_callable: scrape_ntd_ridership_xlsx_url
+# provide_context: true
+# ---
+import logging
+
+import requests
+from bs4 import BeautifulSoup
+from pydantic import HttpUrl, parse_obj_as
+
+from airflow.models import Variable
+
+
+def scrape_ntd_ridership_xlsx_url():
+    # page to find download URL on
+    url = "https://www.transit.dot.gov/ntd/data-product/monthly-module-raw-data-release"
+    req = requests.get(url)
+    soup = BeautifulSoup(req.text, "html.parser")
+
+    # Look for an anchor tag where the href ends with '.xlsx' and starts with '/sites/fta.dot.gov/files/'
+    link = soup.find(
+        "a",
+        href=lambda href: href
+        and href.startswith("/sites/fta.dot.gov/files/")
+        and href.endswith(".xlsx"),
+    )
+
+    # Extract the href if the link is found
+    file_link = link["href"] if link else None
+
+    updated_url = "https://www.transit.dot.gov" + file_link
+    # print('https://www.transit.dot.gov' + file_link)
+
+    validated_url = parse_obj_as(HttpUrl, updated_url)
+
+    logging.info(f"Validated URL: {validated_url}.")
+
+    # Set or overwrite the variable
+    # Will this work?
+    # Variable.set("CURRENT_NTD_RIDERSHIP_URL", f"'{validated_url}'")
+    Variable.set("CURRENT_NTD_RIDERSHIP_URL", validated_url)
```

Comment:
@charlie-costanzo I don't actually know whether this will work (i.e. if the value of the variable will immediately and persistently be available to the rest of the tasks). I don't hate it, but for what it's worth, the [documented Airflow recommendation](https://airflow.apache.org/docs/apache-airflow/stable/core-concepts/variables.html#:~:text=Variables%20are%20global%2C%20and%20should%20only%20be%20used%20for%20overall%20configuration%20that%20covers%20the%20entire%20installation%3B%20to%20pass%20data%20from%20one%20Task/Operator%20to%20another%2C%20you%20should%20use%20XComs%20instead.) is to use XComs for this purpose. Have you had a chance to look into that?

Reply:
just switched to using xcoms if you're able to take a look